src/graphnet/data/dataset.py

```diff
@@ -598,36 +598,51 @@ def _create_graph(
 
         # Additionally add original features as (static) attributes
         for index, feature in enumerate(graph.features):
-            graph[feature] = graph.x[:, index].detach()
+            if feature not in ["x"]:
+                graph[feature] = graph.x[:, index].detach()
 
         # Add custom labels to the graph
         for key, fn in self._label_fns.items():
             graph[key] = fn(graph)
-
         return graph
 
     def _get_labels(self, truth_dict: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
We really ought to reconsider this function, and same for the double-bang one...

Reply:
Yes, I agree. Maybe in the future, we could make a collection of `Labels` that the `Dataset` just defaults to.